psi4/driver/p4util/python_helpers.py

```diff
@@ -661,14 +817,16 @@ def _core_wavefunction_set_variable(cls, key, val):
             raise ValidationError("psi4.core.Wavefunction.set_variable: Target variable " + key +
                                   " already a scalar variable!")
         else:
-            cls.set_array_variable(key, core.Matrix.from_array(val))
+            cls.set_array_variable(key, core.Matrix.from_array(_qcvar_reshape_set(key, val)))
     else:
         if cls.has_array_variable(key):
             raise ValidationError("psi4.core.Wavefunction.set_variable: Target variable " + key +
                                   " already an array variable!")
         else:
             cls.set_scalar_variable(key, val)
 
+    # TODO _qcvar_warnings(key)
```

Comment:
Are these todos still todos?

Reply:
regrettably, yes. I'd like to have the qcvar_warnings active to fend off downstream API setting of the QUAD XY vars. unfortunately, there's some internal call I can't find, so I can't make psi4 itself clean, hence the todo.